lib/print-config.js

```diff
@@ -0,0 +1,382 @@
+const parser = require('yargs-parser')
+
+/**
+ * Function: printConfig
+ *
+ * @param {Object} yargs: instance of populated yargs object.
+ * @param {Function} hideInstrumenteeArgs: Callback defined in lib/parse-args.
+ * @returns {undefined}
+ *
+ * Entry point for print config logic from lib/parse-args.js file.
+ * Kills process at the end of execution.
+ *
+ */
+function printConfig (yargs, hideInstrumenteeArgs) {
+  const argv = process.argv.slice(2)
+  const checkArgs = parser(argv)
+
+  let shouldPrint = false
+
+  if (Object.keys(checkArgs).includes('print-config')) {
+    // checkArgs['print-config'] could contain a boolean or a string
+    // representing a boolean.
+    if (typeof checkArgs['print-config'] === 'boolean') {
+      shouldPrint = checkArgs['print-config']
+    } else if (typeof checkArgs['print-config'] === 'string') {
+      shouldPrint = JSON.parse(checkArgs['print-config'])
+    }
+  }
+
+  if (shouldPrint) {
+    const args = yargs.parse(hideInstrumenteeArgs())
+    const cmdExecuted = 'c8 ' + argv.join(' ')
+    const cleanArgs = cleanUpArgumentObject(args)
+
+    if (args.printConfigFormat === 'text') {
+      printConfigText(cleanArgs, cmdExecuted)
+    } else if (checkArgs.printConfigFormat === 'json') {
+      const jsonYargs = JSON.stringify(cleanArgs, 2)
+      console.log(jsonYargs)
+    }
+
+    // DO NOT REMOVE!  This is intentional.
```

Comment:
Could you expand on the comment to explain why this `process.exit()` is necessary?

Why is this `process.exit()` necessary?

Reply:
The reason for this `process.exit()` is that it stops c8 from producing a coverage report after showing the configuration report.  Additionally, this logic allows JSON to be also printed, and possibly consumed by another process.

@bcoe can you anticipate any use cases where a configuration report and coverage report are necessary?